internal/zinc-compile-core/src/test/scala/sbt/internal/inc/javac/JavaCompilerSpec.scala

```diff
@@ -71,7 +79,7 @@ class JavaCompilerSpec extends UnitSpec with DiagrammedAssertions {
   def findsErrors(compiler: XJavaTools) = {
     val (result, problems) = compile(compiler, Seq(knownSampleErrorFile), Seq("-deprecation"))
     assert(result == false)
-    assert(problems.size == 6)
+    assert(problems.size >= 5)
```

Comment:
Can we have two different asserts depending on the JDK the tests run? If JDK8 or 9 go for 6 errors, otherwise go for 5.

Reply:
ok.

Reply:
https://github.com/sbt/zinc/pull/640/commits/23485a8fa5ebc208805727c1121f9126874022dd

```scala
    assert(problems.size == {
      sys.props("java.specification.version") match {
        case "1.8" | "9" => 6
        case _           => 5
      }
    })
```